R/config.R

```diff
@@ -335,23 +335,18 @@ python_config <- function(python, required_module, python_versions, forced = NUL
     python_libdir_config <- function(var) {
       python_libdir <- config[[var]]
       ext <- switch(Sys.info()[["sysname"]], Darwin = ".dylib", Windows = ".dll", ".so")
-      libpython <- file.path(python_libdir, paste0("libpython" , version, c("", "m"), ext))
-      libpython_exists <- libpython[file.exists(libpython)]
-      if (length(libpython_exists) > 0)
-        libpython_exists[[1]]
-      else
-        libpython[[1]]
+      pattern <- paste0("^libpython", version, "m?", ext)
+      libpython <- list.files(python_libdir, pattern = pattern, full.names = TRUE)
     }
-    if (!is.null(config$LIBPL)) {
-      libpython <- python_libdir_config("LIBPL")
-      if (!file.exists(libpython)) {
-        if (!is.null(config$LIBDIR))
-          libpython <- python_libdir_config("LIBDIR")
-        else
+    for (libsrc in c("LIBPL", "LIBDIR")) {
+      if (!is.null(config[[libsrc]])) {
+        libpython <- python_libdir_config(libsrc)[1]
+        if (is.na(libpython))
```

Comment:
When could this be `NA`? Won't `list.files()` always return a character vector of length >= 0?

Reply:
Oh, because indexing a zero-length character will return `NA`, e.g.

``` r
> character()[1]
[1] NA
```

I think it would be more clear to just use e.g.

``` r
libpython <- python_libdir_config(libsrc)[1]
if (length(libpython)
  break
```

and then use the first entry of `libpython` downstream as appropriate. But the existing code should be correct too.

Reply:
Good idea, I'll update this pull.